### PR TITLE
DOCS-8120 Spec Downloader Lightning Component

### PR DESCRIPTION
**Pull request by wwarner-salesforce**
DOCS-8120 Spec Downloader Lightning Component

### CODE REVIEW

**Review by ferferrarazzo — APPROVED** [review]
LGTM

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 1]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
 Registrations Require Approval:: Select whether registration requires approval.
 Approver ID:: Specifies the approver ID.
 
+== Spec Downloader
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc:376]: Spec Downloader Versions and Options?

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 2]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
...
+
+The Spec Downloader component enables users to download an API specification.
+
+The API specification downloaded from API Community Manager is the same as the specification downloaded from Anypoint Exchange.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc:380]: How about...:
```suggestion
The Spec Downloader component enables users to download an API specification.

The Spec Downloader component <of this> enables users to download an API specification from API Community Manager that is the same as the Anypoint Exchange specification. In <this UI page>, users can click either of the following:
* *Download Spec as RAML* +
* *Download Spec as OAS* +
```
...but I can't figure out what the rest means. Is the general practice to download to the same type (RAML or OAS) as originally uploaded? And if the two differ, then "the downloaded version is automatically generated"? What does that mean, that it is automatically generated; what is the alternative? And why would an automatically generated version be different from its source?

If this is clear to your intended audience, then ignore my confusion. But look at this text again and be sure that the wording you use "connects the dots" and does not rely on audience assumptions.

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 3]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
...
+
+The available buttons are *Download Spec as RAML* and *Download Spec as OAS*. If the API specification is uploaded as RAML and downloaded as OAS, or uploaded as OAS and downloaded as RAML, then the downloaded version is automatically generated and may have small inconsistencies with the uploaded version.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: See above.

Comment (kbostrom) [reply modules/ROOT/pages/acm-lightning-components.adoc]: may > might

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 4]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
...
+
+You can specify values for the following parameters of the Spec Downloader component:
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc:386]: So, after the user chooses the download type, the user customizes using parameters?
Or is this what the admin does?
How are these parameters related to the previous text about the buttons? Do we need a screenshot?
If these are UI labels, they should be formatted as such (* *).

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 5]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
...
+
+Record Id:: Specifies the API. Default is `{!recordId}`, which downloads the specification of the API selected on the current page.
+Layout:: Specifies the layout of the buttons, `Horizontal` or `Vertical`. Default is `Horizontal`.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
Layout:: Specifies the layout of the buttons, `Horizontal` (default) or `Vertical`.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 6]
```diff
@@ -373,6 +373,19 @@ Extra Field Set:: Specifies extra fields.
...
+
 == Filter Conditions [[filter-conditions]]
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc:391]:
```suggestion
== Filter Conditions Parameter [[filter-conditions]]
```

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 1 above)
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc:376]:
```suggestion
== Spec Downloader Component
```

**Review by kbostrom — APPROVED** [review]
Edited

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 1 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc:376]: On this page each header is the exact name of the component, such as "Spec Downloader", so I'm preserving the current text there.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 2 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc:380]: This is explained in more detail in the Exchange docs, and will be somewhat clearer to the audience, but it's a good point, and I added some more explanation. Users can upload either type, and the other type will be generated automatically so that both are available. The alternative would be for one of them to be missing. The differences from the source are caused by unavoidable differences in the RAML and OAS formats.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 3 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Good point. Fixed.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 4 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc:386]: ACM uses Salesforce Community Cloud, which is an interactive web page for building interactive web pages. The admin uses ACM to drag and drop the components listed on this page into a page for developer users and then clicks "publish" to build it. So the admin will drag and drop a Spec Downloader onto the page, and specify the Record Id and Layout options, and the users will just see buttons to click: the *Download Spec as RAML* or *Download Spec as OAS* buttons. "Vertical" and "Horizontal" are options in a drop-down menu.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 5 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Good point! Fixed.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 6 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc:391]: This one is also an exact component name, so I'm preserving the current text.
